src/NServiceBus.Core/Unicast/RunningEndpointInstance.cs

```diff
@@ -26,6 +27,8 @@ public async Task Stop(CancellationToken cancellationToken = default)
                 return;
             }
 
+            stoppingTokenSource.Cancel();
```

Comment:
Should this go behind the semaphore and status check? It seems somewhere after `status = Status.Stopping` on line 42 would be suitable?

Reply:
Are you worried about this operation being idempotent? Looking at ILSpy it seems it is (or enough, anyway) so why not call it here?

Reply:
Discussed on the sync and since its idempotent, we will leave as is